google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -137,9 +138,10 @@ public CancellableRunnable createDeployRunner(
         wrapCallbackForUsageTracking(
             callback, deploymentConfiguration, source.getFile(), targetEnvironment));
 
-    if (targetEnvironment == AppEngineEnvironment.APP_ENGINE_STANDARD) {
+    if (targetEnvironment.isStandard()
+        || (targetEnvironment.isFlexible() && AppEngineUtil.isFlexCompat(project, source))) {
```

Comment:
we have a real terminology problem around identifying 'standard framework' vs 'standard environment'.